lib/seattleflu/id3c/cli/command/etl/redcap_det_childcare.py

```diff
@@ -0,0 +1,574 @@
+        return None
+
+    # Create the participant resource entry and reference.
+    # Assumes that the project language is the participant's preferred language.
+    patient_entry, patient_reference = create_patient_using_demographics(
+        sex = enrollment['core_sex'],
+        preferred_language = LANGUAGE_CODE[enrollment.project.id],
+        first_name = enrollment['core_participant_first_name'],
+        last_name = enrollment['core_participant_last_name'],
+        birth_date = enrollment['core_birthdate'],
+        zipcode = enrollment['core_zipcode'],
+        record = enrollment,
+        system_identifier = INTERNAL_SYSTEM)
+
+    if not patient_entry:
+        LOG.warning('Skipping record with insufficient information to construct patient')
+        return None
+
+    birthdate = parse_date_from_string(enrollment.get('core_birthdate'))
+
+    location_resource_entries = build_location_resources(
+        db = db,
+        cache = cache,
+        housing_type = enrollment.get('core_housing_type'),
+        primary_street_address = enrollment['core_home_street'],
+        secondary_street_address = enrollment['core_apartment_number'],
+        city = enrollment['core_home_city'],
+        state = enrollment['core_home_state'],
+        zipcode = enrollment['core_zipcode'],
+        system_identifier = INTERNAL_SYSTEM)
+
+    persisted_resource_entries = [patient_entry, *location_resource_entries]
+
+    for redcap_record_instance in redcap_record_instances:
+
+        event_type = None
+        study_arm = None
+
+        if redcap_record_instance.event_name.startswith(ENROLLMENT_EVENT_NAME_PREFIX):
+            event_type = EventType.ENROLLMENT
+        elif redcap_record_instance.event_name.startswith(ENCOUNTER_EVENT_NAME_PREFIX):
+            event_type = EventType.ENCOUNTER
+        else:
+            LOG.error(f'The record instance has an unexpected event name: {redcap_record_instance.event_name}')
+            continue
+
+        if '_arm_1' in redcap_record_instance.event_name:
+            study_arm = StudyArm.PRIMARY
+        elif '_arm_2' in redcap_record_instance.event_name:
+            study_arm = StudyArm.SECONDARY
+        else:
+            LOG.error(f'The record instance has an unexpected study arm in the event name: {redcap_record_instance.event_name}')
+            continue
+
+        # Skip an ENCOUNTER instance if we don't have the data we need to
+        # create an encounter. Require the participant to have provided
+        # survey data or a sample.
+        if event_type == EventType.ENCOUNTER \
+            and not is_complete('symptom_check', redcap_record_instance) \
+            and not is_complete('swab_kit_reg', redcap_record_instance) \
+            and not is_complete('post_collection_data_entry_qc', redcap_record_instance):
+            LOG.debug('Skipping record instance with insufficient information to construct the encounter')
```

Comment:
I think we still want to ingest sample even if the kit reg is not complete as long as the PCDEQC form is complete. The lab will accept samples that have the PT name and collection date on the tube even if the kit reg is not complete.

Reply:
Exactly.
This skips a record instance if all these are true:
symptom_check is not complete
swab_kit_reg is not complete
post_collection_data_entry_qc is not complete
That is, if if one of those 3 is complete it will create the encounter.

This is the 4th encounter in test record 21. (The test records are getting deleted soon, but that's where I tested it.)

Reply:
Ah, right. My brain somehow turned around the logic. Thanks for clarifying!